TimberAPI/AssetLoaderSystem/ResourceAssetPatch/TimberApiResourceAssetLoader.cs

```diff
@@ -11,15 +10,30 @@ public class TimberApiResourceAssetLoader : IResourceAssetLoader
     {
         private readonly IAssetLoader _assetLoader;
 
-        private Shader _shader;
+        private static Shader _buidlingShader;
+        private static Shader _dirtShader;
+        private static Shader _pathShader;
 
-        private Shader Shader
+        private static Shader Shader(Material mat)
```

Comment:
Now that this is a method and not a property, I feel like renaming it would be good, for example "GetShader()"